File: 0101-0200/0125.Valid_Palindrome.java
//1ms


class Solution {
    public boolean isPalindrome(String s) {
        char[] chars=new char[s.length()];
        int i=0;
        for (char c: s.toCharArray())
        {
            if ((c>='a'&&c<='z')||(c>='0'&&c<='9'))
            {
                chars[i]=c;
                i++;
            }
            if (c>='A'&&c<='Z')
            {
                chars[i]=(char)(c+32);
                i++;
            }
        }
        for (int j=0; j<i; j++)
        {
            if (chars[j]!=chars[i-j-1])
                return false;
        }
        return true;
    }
}

File: 0101-0200/0125.Valid_Palindrome.rs
//0ms


impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let mut iter = s
            .chars()
            .filter(|c| c.is_alphanumeric())
            .map(|c| c.to_ascii_lowercase());

        iter.clone().eq(iter.rev())
    }
}
